tests/test-admin-deploy-2.sh

```diff
@@ -63,3 +63,34 @@ assert_has_dir sysroot/boot/ostree/testos-${bootcsum}
 assert_file_has_content sysroot/ostree/deploy/testos/deploy/${newrev}.0/etc/os-release 'NAME=TestOS'
 
 echo "ok manual cleanup"
+
+assert_n_pinned() {
+    local n=$1
+    ${CMD_PREFIX} ostree admin status > status.txt
+    local n_pinned="$(grep -F -c -e 'Pinned: yes' < status.txt)"
+    if test "${n_pinned}" '!=' "${n}"; then
+        cat status.txt
+        fatal "${n_pinned} != ${n}"
+    fi
+}
+assert_n_pinned 0
+${CMD_PREFIX} ostree admin pin 0
+assert_n_pinned 1
+${CMD_PREFIX} ostree admin pin -u 0
+assert_n_pinned 0
+echo "ok pin unpin"
+
+${CMD_PREFIX} ostree admin pin 0
+${CMD_PREFIX} ostree admin pin 1
+assert_n_pinned 2
+os_repository_new_commit
+${CMD_PREFIX} ostree admin upgrade --os=testos
+assert_n_pinned 2
```

Comment:
Maybe let's explicitly also check that the pinned deployments are still there? (I.e. just grep for checksums? Or maybe just checking that the deployment count went up to three.)